spring-kafka/src/main/java/org/springframework/kafka/listener/TimestampedException.java

```diff
@@ -34,14 +34,36 @@
 
 	private final long timestamp;
 
+	@Deprecated
 	public TimestampedException(Exception ex, Clock clock) {
 		super(ex.getMessage(), ex);
 		this.timestamp = Instant.now(clock).toEpochMilli();
 	}
 
+	@Deprecated
```

Comment:
And what is wrong with the convenient ctor taking the `now()`  for its timestamp?
Why do we need to deprecate this one?